### PR TITLE
Update API to match with updated aarch64 API / add auth functions

### PR DESCRIPTION
**Pull request by iojcde**
https://github.com/fosshostorg/aarch64/pull/68 
The above PR updated the aarch API. 

This PR adds authentication functions to the library, and updates the API methods.

### CODE REVIEW

**Comment by iojcde** [comment]
Sorry for the many commits, I could squash them if you want.

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/jcdea/aarch64-client-go
```
Comment (natesales) [inline go.mod:1]: This is the wrong package URI, it should be `github.com/natesales/aarch64-client-go`

File: main.go [hunk 2]
```diff
@@ -57,6 +60,11 @@ type ProjectsResponse struct {
 	Projects []Project `json:"data"`
 }
 
+func NewClient(APIKey string) Client {
+	client := &http.Client{}
+	return Client{APIKey: APIKey, client: client}
```
Comment (natesales) [inline main.go:65]: Minor thing: there's no need to assign the `http.Client` pointer to a variable. You can just include it inline in the `Client` struct.

**Review by natesales — CHANGES_REQUESTED** [review]

**Comment by natesales** [comment]
Thanks for the PR, just a few minor things to address and I'll merge.

File: go.mod (hunk 1 above)
Comment (iojcde) [reply go.mod:1]: Oop sorry, I forgot to change the package url

**Comment by iojcde** [comment]
@natesales I'll just close this PR and open a new one.

**Pull request closed without merging**